api/src/test/java/org/pmiops/workbench/api/DataSetControllerTest.java

```diff
@@ -320,7 +329,32 @@ private void stubGetWorkspace(String ns, String name, String creator,
     );
   }
 
+  private List<DomainValuePair> mockDomainValuePair() {
+    List<DomainValuePair> domainValues = new ArrayList<>();
+    DomainValuePair domainValuePair = new DomainValuePair();
+    domainValuePair.setDomain(Domain.CONDITION);
+    domainValuePair.setValue("PERSON_ID");
+    domainValues.add(domainValuePair);
+    return domainValues;
+  }
 
+  private void mockLinkingTableQuery(ArrayList<String> domainBaseTables) {
+    TableResult tableResultMock = mock(TableResult.class);
+    ArrayList<FieldValueList> values = new ArrayList<>();
+    domainBaseTables.forEach(domainBaseTable -> {
+      ArrayList<Field> schemaFields = new ArrayList<>();
+      schemaFields.add(Field.of("OMOP_SQL", LegacySQLTypeName.STRING));
+      schemaFields.add(Field.of("JOIN_VALUE", LegacySQLTypeName.STRING));
+      FieldList schema = FieldList.of(schemaFields);
+      ArrayList<FieldValue> rows = new ArrayList<>();
+      rows.add(FieldValue.of(FieldValue.Attribute.PRIMITIVE, "PERSON_ID"));
+      rows.add(FieldValue.of(FieldValue.Attribute.PRIMITIVE, domainBaseTable));
+      FieldValueList fieldValueList = FieldValueList.of(rows, schema);
+      values.add(fieldValueList);
+    });
+    doReturn(values).when(tableResultMock).getValues();
+    doReturn(tableResultMock).when(bigQueryService).executeQuery(any());
```

Comment:
@calbach FYI, I figured out how to mock the execute query call. Of note, this does _not_ handle constructing the actual query job param, so this will hit everything :( but I think that is doable. (Just not something I wanted to tackle here until we need to.)